ee/clickhouse/views/person.py

```diff
@@ -1,56 +1,62 @@
-from rest_framework import request, response
+from typing import Callable, Dict, Optional, Tuple
+
 from rest_framework.decorators import action
 from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
 
 from ee.clickhouse.client import sync_execute
 from ee.clickhouse.models.person import delete_person
 from ee.clickhouse.queries.clickhouse_retention import ClickhouseRetention
 from ee.clickhouse.queries.clickhouse_stickiness import ClickhouseStickiness
-from ee.clickhouse.queries.funnels.funnel_persons import ClickhouseFunnelPersons
-from ee.clickhouse.queries.funnels.funnel_trends_persons import ClickhouseFunnelTrendsPersons
+from ee.clickhouse.queries.funnels import ClickhouseFunnelPersons, ClickhouseFunnelTrendsPersons
 from ee.clickhouse.queries.trends.lifecycle import ClickhouseLifecycle
 from ee.clickhouse.sql.person import GET_PERSON_PROPERTIES_COUNT
 from posthog.api.person import PersonViewSet
 from posthog.api.utils import format_next_absolute_url, format_next_url
+from posthog.constants import INSIGHT_FUNNELS, FunnelVizType
+from posthog.decorators import cached_function
 from posthog.models import Event, Filter, Person
 
 
-# TODO: Move grabbing all this to Clickhouse. See WIP-people-from-clickhouse branch.
 class ClickhousePersonViewSet(PersonViewSet):
 
     lifecycle_class = ClickhouseLifecycle
     retention_class = ClickhouseRetention
     stickiness_class = ClickhouseStickiness
 
-    @action(methods=["GET"], detail=False)
-    def funnel(self, request: request.Request, **kwargs) -> response.Response:
+    @action(methods=["GET", "POST"], detail=False)
```

Comment:
POST is important, because GET has issues with the URL being too long (on /insight/funnel/) and persons take all the same parameters!

Reply:
I'm tempted to remove GET all together, since this is a new API endpoint.